frontend/src/hooks/useGenerateReportEffect.ts

```diff
@@ -1,118 +1,324 @@
+import { ReportCallbackResponse, ReportResponseDTO } from '@src/clients/report/dto/response';
 import { exportValidityTimeMapper } from '@src/hooks/reportMapper/exportValidityTime';
 import { DATA_LOADING_FAILED, DEFAULT_MESSAGE } from '@src/constants/resources';
-import { ReportResponseDTO } from '@src/clients/report/dto/response';
+import { IPollingRes, reportClient } from '@src/clients/report/ReportClient';
+import { DateRange, selectConfig } from '@src/context/config/configSlice';
 import { ReportRequestDTO } from '@src/clients/report/dto/request';
-import { reportClient } from '@src/clients/report/ReportClient';
+import { formatDateToTimestampString } from '@src/utils/util';
 import { TimeoutError } from '@src/errors/TimeoutError';
 import { METRIC_TYPES } from '@src/constants/commons';
+import { useAppSelector } from '@src/hooks/index';
 import { useRef, useState } from 'react';
+import get from 'lodash/get';
 
-export interface useGenerateReportEffectInterface {
+export type PromiseSettledResultWithId<T> = PromiseSettledResult<T> & {
+  id: string;
+};
+
+export interface IUseGenerateReportEffectInterface {
```

Comment:
rename IUseGenerateReportEffectInterface to IUseGenerateReportEffect

Reply:
done